Tests/SwiftParserTest/Attributes.swift

```diff
@@ -0,0 +1,21 @@
+@_spi(RawSyntax) import SwiftSyntax
+@_spi(RawSyntax) import SwiftParser
+import XCTest
+
+final class AttributeTests: XCTestCase {
+  func testMissingArgumentToAttribute() {
+    AssertParse(
+      """
+      @_dynamicReplacement(#^DIAG_1^#
+      func #^DIAG_2^#test_dynamic_replacement_for2() {
+      }
+      """,
+      diagnostics: [
+        // FIXME: We should be complaining about the missing ')' for the attribute
+        DiagnosticSpec(locationMarker: "DIAG_1", message: "Expected 'for'"),
+        DiagnosticSpec(locationMarker: "DIAG_1", message: "Expected ':'"),
+        DiagnosticSpec(locationMarker: "DIAG_2", message: "Expected ')'"),
```

Comment:
Obviously not for this PR and even somewhat unrelated to these diagnostics - but should we limit the expected diagnostic to one per location? Not sure it's entirely useful to have both `for` and `:` here.

Reply:
I’m not sure it that’s easy to enforce. We only resolve source locations from syntax nodes when printing the diagnostics and before that, these diagnostics are anchored to different nodes (the missing `for` token and the missing `:` token). So the only place I think we could realistically enforce this would be `AssertParse`.